code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -35,7 +35,7 @@
 		if(cycles >= 4)
 			for(var/mob/living/simple_animal/K in oview(src, 1)) //Only for corpse right next to/on same tile
 				if(K.stat == DEAD || (!K.check_death_method() && K.health <= HEALTH_THRESHOLD_DEAD))
-					visible_message("<span class='danger'>[src] consumes [target] whole!</span>")
+					visible_message("<span class='danger'>[src] consumes [K.name] whole!</span>")
```

Comment:
just `K` will do here. It will auto make a fitting name. Usually it'll be the same as `K.name`